### DISCUSSION TITLE
Not receiving SMS through Klaviyo integration

### DISCUSSION THREAD

**Discussion by palash-cj** [saleor/saleor#12150] [General]
Hi there!
I want to receive SMS for order confirmation, fulfillment etc. For this I integrated Klaviyo in saleor dashboard. The steps I followed are as follow:
1. I selected the "Apps" feature from saleor-dashboard sidebar.
2. Installed klaviyo app using manifest url: https://klaviyo.saleor.app/api/manifest
3. Inserted PUBLIC_TOKEN that I got from Klaviyo account.
4. Configured all the metrics required to activate SMS for.

Now I am able to track the order creation by customers in klaviyo dashboard but not receiving SMS for the same. Please help me out regarding this.

#Klaviyo #saleor #saleor-dashboard #development #community

**Comment by lkostrowski** [comment]
Hey @palash-cj 

Klaviyo app is only sending metrics to Klaviyo itself - configuring SMS or other messages needs to be done on your side in Klaviyo dashboard.

If you see your metric in Klaviyo, most likely app is working fine and you need to check your workflow in Klaviyo itself.

If you have further problems please open issues in https://github.com/saleor/apps

**Reply by palash-cj** [reply]
Thanks @lkostrowski for your answer.

In the dashboard I have configured the following metrics:
SALEOR_CUSTOMER_CREATED_METRIC
SALEOR_FULFILLMENT_CREATED_METRIC
SALEOR_ORDER_CREATED_METRIC
SALEOR_ORDER_FULLY_PAID_METRIC
These are visible in the Klaviyo dashboard. 

When I am creating new customer from dashboard, its getting fetched at Klaviyo dashboard. But customer's phone number is getting blank value as while creating customer there is no option for customer phone number. Rather in the primary address section there is phone number field, which is not being used while creating customer. Please guide me how to insert phone number from saleor to klaviyo.

I activated sms services and tollfree no is also verified. Now I created template for customer creation SMS. Please suggest me how can I integrate template id in saleor for receiving sms.

Also, do flow creation needed to receive sms?

**Reply by lkostrowski** [reply]
Ok, I get your issue now. I will try to check it this week and try to add additional fields to the payload

You can also open the PR on your own, likely here:
https://github.com/saleor/apps/blob/main/apps/klaviyo/pages/api/webhooks/customer-created.ts

**Reply by palash-cj** [reply]
Thanks @lkostrowski
It seems like, payload is missing the customer details like first name, last name, phone number, address, within the code. Also, we don't have access to the same code as the Klaviyo is a third party app and installs with its manifest url.

It would more helpful if you could add those fields in the code.

**Reply by lkostrowski** [reply]
Saleor Apps are open source, so developers can always contribute and apps will be shipped to the Appstore built-in :) 

However, I will try to extend this payload, it should be a quick fix

**Reply by lkostrowski** [reply]
Hey @palash-cj, I checked current implementation

In Saleor, user creation doesn't require phone number (and actually many other fields). Phone number is connected to the address.

To send an SMS to the user, you can:
1. Enforce default billing/shipping address when user is created (https://docs.saleor.io/docs/3.x/api-reference/inputs/user-create-input)
2. Add a metadata  field with a phone number, to hack it around

Current Klaviyo App implementation is sending addresses to Klaviyo, so lets try to create a user with address and you should be able to use it in Klaviyo

**Reply by palash-cj** [reply]
@lkostrowski Current Klaviyo App implementation is not sending anything to Klaviyo except email.

 I tried to create a customer through saleor-dashboard with all the fields including Primary address. And in the Klaviyo dashboard that profile is visible but with no user details. Contact section within the profile is blank.

**Reply by lkostrowski** [reply]
Can you check what exact the payload of the metric in Klaviyo is (inside Klaviyo dashboard)?

**Reply by palash-cj** [reply]
Metric payload is not visible directly in the Klaviyo dashboard but activity feed is showing the data as per the graphQL query in this file https://github.com/saleor/apps/blob/main/apps/klaviyo/pages/api/webhooks/customer-created.ts
